### ISSUE TITLE
ENA submission error (FTP server)

### ISSUE THREAD

**Issue by BettinaVgn** [DKMS-LSL/typeloader#40]
Hey there,
I am new to TypeLoader. I installed the software recently and am currently testing it using a test account. I was able to create projects and successfully load new alleles. I wanted to test ENA submission now but am prompted with the following error: 
__________________________________
ENA submission error Submission to ENA failed even though validation passed:

ERROR: ENA rejected your files (submission failed):

WARN : Retrying connecting to FTP server.
WARN : Retrying connecting to FTP server.
WARN : Retrying connecting to FTP server.
WARN : Retrying connecting to FTP server.
WARN : Retrying connecting to FTP server.
WARN : Retrying connecting to FTP server.
ERROR: Failed to connect to webin.ebi.ac.uk using FTP. Failed to upload files to webin.ebi.ac.uk because of a system error.
ERROR: Failed to connect to webin.ebi.ac.uk using FTP. Failed to upload files to webin.ebi.ac.uk because of a system error.

The complete submission has been rejected.

Submission aborted. Please fix the values and try again!
__________________________________
FTP user and password should be fine because I can login here without problems: https://www.ebi.ac.uk/ena/submit/webin/login
I also checked with our IT that it's ok to leave the proxy server field empty because I was told this. Could you please help me to identify the problem?

Thanks in advance and best,
Bettina

**Comment by bmschoene** [comment]
Hi Bettina, thanks for trying TypeLoader, and for reaching out. 

"System Error" sounds like the problem is on ENA's side (maybe their server was temporarily unavailable).
Can you please try again?

If the problem still persists, please upload a log file here so I can investigate further? You can get this file via `Options` => `Download Log File`. Please heed the instructions in the popup dialogs.
Thank you.

**Comment by BettinaVgn** [comment]
Hey Bianca,
thank you for your reply! I have tried it on 3 different days now and it still doesn't work. Please find today's log:
[20240306_103520.log](https://github.com/DKMS-LSL/typeloader/files/14507939/20240306_103520.log)
Hope this helps! 

Thanks and best,
Bettina

**Comment by bmschoene** [comment]
Thanks.

Hm. That sadly doesn't tell me more than what we previously had. And it's nothing I can fix, because the problem is clearly on ENA's side, or somewhere between you and ENA. I can access their servers fine from here.

What you can try:
- Go to your TypeLoader `Settings` => `Preferences` => increase the number for `ENA timeout after x seconds` (I don't think it will change things, as there is no TimeoutError in your log, but it's always worth a try).
- Find the file `C:\TypeLoader_Data\test\projects\20240306_T_HLA-A_NAA24\.\webin-cli.report` on your computer and check whether it contains more specific information than "Failed to upload files to webin.ebi.ac.uk because of a system error."

If all of this does not work, you'll have to ask ENA for support. Apparently, [other people have experienced the same problem earlier this year](https://github.com/enasequence/webin-cli/issues/112), but I don't know what became of that. You can either open a ticket there, or at ENA help desk. Both take time. If you don't get an answer after 2 weeks or so, it can help to escalate via email to their head of department Tony Burdett. (That's what my head of department did the last time I really needed something fixed in a timely fashion and was getting no response for weeks.)

You could also manually upgrade your ENA Webin CLI plugin, just to be safe. (But this really should not be the source of your problem, as the old plugin should work until April 8th (and still does from here)). To do that, you would go to [https://github.com/enasequence/webin-cli/releases](https://github.com/enasequence/webin-cli/releases) and download the file **[webin-cli-7.0.1.jar]**. Save it in the directory `C:\Program Files (x86)\TypeLoader\lib\typeloader2\ENA_Webin_CLI` and then try again. (You can leave the existing version there. TypeLoader always automatically uses the highest version it finds in this directory.) 
However, since they - once again *sighs* - changed their interface, **the current TypeLoader does not quite work with the new webin-CLI version**, so you will get an error there, anyway. I'm working on a new TypeLoader release that will incorporate and correctly handle the new webin-CLI version, but it will be a few weeks until then.

Sorry I can't be of more help.
Let me know of any progress?

**Comment by BettinaVgn** [comment]
Hey Bianca,
thanks again for all you help!

So, as you suspected:
-increasing the ENA timeout did not help
-upgrading the ENA Webin CLI plugin did not change anything
-the webin-cli.report did not give me any notable information but maybe you can take a look, since your the expert: 
[PRJEB73744_20240308085246_webin-cli.report.txt](https://github.com/DKMS-LSL/typeloader/files/14536766/PRJEB73744_20240308085246_webin-cli.report.txt)

If this does not give you any specific hint I will create a ticket at ENA helpdesk and hope for a fast reply as you suggested. Let's see what happens..

Best,
Bettina

**Comment by bmschoene** [comment]
Thanks. No, I don't see anything helpful in there, either. But maybe the ENA people will; best attach this to the ticket.

The new TypeLoader version should be out before April 8th. Maybe they'll have fixed their problem until then...
(Sadly, ENA appears to be chronically understaffed, so responses from them often take very long.)

**Comment by bmschoene** [comment]
Following up on this: Have you had any response from ENA? Or any luck with new attempts?
The new release is up (which contains the new webin CLI version), in case this should help.

**Comment by BettinaVgn** [comment]
Hey there!  Thanks for checking in!

Sadly, ENAs response has not been a lot of help. They suggested to check the following things:
"1. You have tried resetting your password
2. It has been at least 15 minutes since you last reset your password
3. Your firewall is configured so as to conform with the below:
https://protect2.fireeye.com/v1/url?k=31323334-501cfaf3-313580f2-454445554331-db408df631c9c9dc&q=1&e=8c453679-e997-48c2-bfa8-f5116085a345&u=https%3A%2F%2Fena-docs.readthedocs.io%2Fen%2Flatest%2Fsubmit%2Ffileprep%2Fupload.html%23
appendix-configuring-your-firewall-for-ena-upload
4. Your upload approach matches one of the instructions available at:
https://protect2.fireeye.com/v1/url?k=31323334-501cfaf3-313580f2-454445554331-db408df631c9c9dc&q=1&e=8c453679-e997-48c2-bfa8-f5116085a345&u=https%3A%2F%2Fena-docs.readthedocs.io%2Fen%2Flatest%2Fsubmit%2Ffileprep%2Fupload.html%23
file-upload-options"

I tried 1-2 and this did not help. I am not sure if 3 is really an issue since our IT confirmed that I can leave proxy server field empty and this should be fine. 4 should also not be an issue, since I am using TypeLoader and the software works in your case..

I also tried the new Typeloader V2.15.0 including Webin-CLI version 7.1.1 but the error is still the same.

I am not really sure what else I can do except report back to ENA that it still doesn't work..

If you have any new ideas please let me know :) thanks!

**Comment by bmschoene** [comment]
This is really weird.

For some reason, their list under 4. does not include [webin CLI](https://ena-docs.readthedocs.io/en/latest/submit/general-guide/webin-cli.html) anymore, which is what TypeLoader uses. However, this _does_ work at our lab, so I don't see why it shouldn't at your lab...

Oh, huh. I just noticed something in your logfile. When you entered your Center Name (during TypeLoader setup or later, when you configured your ENA credentials), did you put quotes around your company name? You don't need those, TypeLoader does that already. I see two sets of quotes around the company name in your logfile, so that could be the problem.

To fix this, [open your TypeLoader Settings](https://github.com/DKMS-LSL/typeloader/blob/master/user_manual/settings.md) => Company and look for the field called "Company Name (for ENA)". Are there quotes around the company name? If yes, remove them. (Actually, remove any surrounding quotes from all those fields, if you have any. Double quotes, single quotes, whatever.) Then click the "Confirm changes" button and try ENA submission again.

Let me know how that works out?

**Comment by BettinaVgn** [comment]
Hey there,
ok this is strange. There are no quotes in the settings. I tried changing the company name to something without spaces, in case the quotes are somehow added because of this but I can still see the quotes in the new log file and the error is the same as always. Do you not see any quotes in your logfile?

**Comment by bmschoene** [comment]
I just checked; we have the 2 sets of quotes, too, so it's not that. Bugger. That would have been a nice solution.

Then yes, it's back to ENA, I'm afraid. Tell them you're using webin-CLI version 7.1.1 with the following command your firewall is fine (if your IT people confirm this; show them the specifications under 3. to make sure) and you used the following command:

`java -jar C:\Program Files (x86)\TypeLoader\lib\typeloader2\ENA_Webin_CLI\webin-cli-7.1.1.jar -context sequence -manifest C:\TypeLoader_Data\test\projects\20240306_T_HLA-A_NAA24\PRJEB73627_20240306103848_manifest.txt -userName ****@****.com -password **** -centerName "****" -inputDir C:\TypeLoader_Data\test\projects\20240306_T_HLA-A_NAA24 -outputDir C:\TypeLoader_Data\test\projects\20240306_T_HLA-A_NAA24 -test -submit`

And that webin-cli successfully validated the files in question but cannot connect to the server (including the actual error message).

Maybe ask them for an explicit way for your IT people to test the server connection from the command line (because that's where the error says the problem is).

I do think the problem is probably somewhere within your IT. I can't hink of any other explanation. But ENA should give your IT people ways to verify and fix that. 

Fingers crossed!

**Comment by kisun** [comment]
Hi @BettinaVgn I had same errors/warnings as yours. I realized that I was using VPN and when I turned VPN off, those warnings disappeared.

**Comment by BettinaVgn** [comment]
Hey there, 
sorry for not answering for so long.. The last comment was actually quite helpful, the only problem is that our IT infrastructure does normally not allow working without a VPN. Even when working from the office we're usually using VPN and the traffic in and out is strictly controlled.. and apparently something was inhibiting the connection to the FTP server during ENA submission.
Our IT tried different things including changing the policy-based routing but this didn't help either. They just couldn't figure out which route was really "blocked". In the end we now decided to do the process outside of our IT infrastructue. We installed TypeLoader on a remote computer and managed to do the submission via private internet connection. I'm glad it finally worked even though we weren't really able to solve the issue.. 
For me it would be fine to close this issue unless you have any other comments :) thanks so much for all your help!
